cmd/query/app/server.go

```diff
@@ -259,9 +318,9 @@ func createHTTPServer(
 		server.TLSConfig = tlsCfg
 	}
 
-	server.staticHandlerCloser = RegisterStaticHandler(r, telset.Logger, queryOpts, querySvc.GetCapabilities())
+	s.staticHandlerCloser = RegisterStaticHandler(r, telset.Logger, queryOpts, querySvc.GetCapabilities())
```

Comment:
so once again, why couldn't this handler be added in CreateRouter? Then it seems you wouldn't be needing to return router, only handler, and it would allow consolidating most handlers (at least the ones for business logic like tenancy or bearer token) together.

Reply:
@yurishkuro we could register the handler there but then how would we set the closer on the server?